pkg/objcache/objcache.go

```diff
@@ -145,36 +181,50 @@ func (c *Cache) Create(key string, size int64) (w io.WriteCloser, err error) {
 
 // Open - open the in-memory file, returns an in memory read seeker.
 // returns an error ErrNotFoundInCache, if the key does not exist.
-func (c *Cache) Open(key string) (io.ReadSeeker, error) {
-	c.mutex.RLock()
-	defer c.mutex.RUnlock()
-
+func (c *cache) Open(key string) (io.ReadSeeker, error) {
 	// Entry exists, return the readable buffer.
```

Comment:
> same comment here: previous code of keeping Lock and Unlock together was better as separating it apart does not give much advantage as it's not doing any locking calls.
> The advantage is avoiding entire body locking, the inference here is to indicate that we are locking only the map reads/writes. I prefer this approach over other.

Reverted back to the previous style, since we don't need RLock anymore.